docs/glossary.rst

```diff
@@ -0,0 +1,66 @@
+===============================================================================
+XBRL Glossary
+===============================================================================
+
+Axis
+^^^^
+
+Axes are defined in the Taxonomy, and are used as a part of the
+Context to identify a Fact beyond the default
+entity, and Period. Axes can be either typed or explicit. An
+explicit Axis will have all of its allowable values defined in the taxonomy. A
+typed Axis will instead allow a filer to input their own values for the Axis.
+
+Concept
+^^^^^^^
+
+Concepts are defined in the Taxonomy, and describe what sort of Facts can be
+reported in a Filing. They provide a written description, and other metadata that
+the taxonomy creator chooses to include. Not all Concepts will have Facts they are
+directly associated with. This is because Concepts can also be used as a part of a
+Concept tree to define relationships with other Concepts. The root of such a tree
+is a Link Role.
+
+Context
+^^^^^^^
+
+A context provides important information for identifying/interpreting a
+Fact. Every context contains both an entity identifier, and a
+Period. A context can also contain user defined Axes.
+
+Fact
+^^^^
+
+An XBRL fact is considered to be an atomic unit of data. It contains a single value,
+with a name that corresponds to a Concept, and a reference to a
+Context.
+
+Filing
+^^^^^^
+
+An XBRL filing is how data is actually reported. It contains a list of
+Facts, and a list of Contexts. A filing is associated with a Taxonomy that is used
+to interpret facts and their relationships to one another.
+
+Link Role
```

Comment:
Can we give an example of how this hierarchical arrangement shows up in the FERC Form 1 data, and how it's translated into collections of tables? It almost sounds like Link Roles end up defining foreign key relationships in the DB.  Could we use the income statements, or utility plant balances, which are spread across multiple tables and organized hierarchically into many categories and sub-categories, where one kind of record in one table actually represents an aggregation of all records in another table.

Reply:
The hierarchical relationship is shown in the diagram in the parsing section. The relationships they define between Concepts groups them into tables, but does not define any foreign key relationship with other tables. Hopefully this is more clear now